kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/TopicsResource.java

```diff
@@ -137,24 +138,37 @@ public void createTopic(
     String topicName = request.getTopicName();
     Optional<Integer> partitionsCount = request.getPartitionsCount();
     Optional<Short> replicationFactor = request.getReplicationFactor();
+    Map<Integer, List<Integer>> replicasAssignments = request.getReplicasAssignments();
     Map<String, Optional<String>> configs =
         request.getConfigs()
             .stream()
             .collect(Collectors.toMap(ConfigEntry::getName, ConfigEntry::getValue));
 
+    // We have no way of knowing the default replication factor in the Kafka broker.
+    short rf = replicationFactor.orElse((short) 0);
+    if (replicasAssignments != null) {
+      // In case of explicitly specified partition-to-replicas assignments, all partitions should
+      // have the same number of replicas.
+      rf = (short) replicasAssignments.values().iterator().next().size();
+    }
```

Comment:
You can have rf being non mutable if you chain it:

```
replicationFactor.orElse(replicasAssignments.values().stream().map(List::size).findAny().orElse(0))
```

We avoid mutables whenever we can. Also, don't use abbreviations. Prefer full names.

Reply:
Unfortunately this expression goes through `Optional<Integer>` and in order to get back from `Integer` to `short` I need to explicitly unbox then do a narrowing conversion. The result is a mouthful, and I'm not sure the much worse readability is worth the `final` here. I replaced it with a ternary conditional, although it can also be wrestled to work with your suggestion. E.g.
```
final short assumedReplicationFactor = replicationFactor.orElse(
    replicasAssignments.isEmpty() ? 0 : (short) replicasAssignments.values().iterator().next().size());
```
vs
```
final short assumedReplicationFactor = replicationFactor.orElse(
        (short) ((int) (
            replicasAssignments.values().stream().map(List::size).findAny().orElse(0))));
```